### DISCUSSION TITLE
Can you explain how the Hugo site generation works?

### DISCUSSION THREAD

**Discussion by endlessuniverse** [daptin/daptin#124] [Q&A] [answered]
Hello there!

Please check this out for reference:

> A CLI tool that pulls data from Contentful and turns it into markdown files for Hugo and other static site generators. It also includes an express server that can be used for local development and content previews.

https://github.com/ModiiMedia/contentful-hugo

It's extremely flexible to create Hugo sites with leaf and branch page bundles.

Can you have a Hugo site at the root of a domain, the API at a subdomain and the admin panel at a subdirectory or subdomain?

Thank you!

**Comment by artpar** [comment, marked as answer]
Hi @endlessuniverse 

I have the hugo compiler integrated in daptin. To use it what you need to do it create a subsite entry of type "hugo" and upload the appropriate hugo directory structure on the backing store (you can easily start with localhost disk backed store) of the subsite. 

Daptin will compile the hugo template with the content in the directory and expose the compiled site on the domain/subpath which you have configured in the subsite. You will also be able to live edit the content/template directly through apis and update the whole site on run time.

**Reply by endlessuniverse** [reply]
Thank you!
